lib/connection_pool.rb

```diff
@@ -46,7 +46,7 @@ def initialize(options = {}, &block)
 
     options = DEFAULTS.merge(options)
 
-    @size = options.fetch(:size)
+    @size = options.fetch(:size).to_i
```

Comment:
to_i will return 0 if this value is invalid or nil.  Should it use `Integer(...)` instead?

Reply:
I opted for `#to_i` to be as noninvasive as possible. I do agree that it's better to use `Integer(...)` but was wary because it may introduce errors to existing code when they upgrade (perhaps poor test coverage). I do agree that failing fast here is better than unexpected behavior. I'll update the code 👍